src/main/resources/hazelcast-config-schema.json

```diff
@@ -1373,20 +1373,535 @@
+          "type": "object",
+          "description": "Configures the server socket used for the member-client communication.",
+          "additionalProperties": false,
+          "$ref": "#/definitions/SocketEndpointConfig"
+        },
+        "wan-server-socket-endpoint-config": {
+          "type": "object",
+          "description": "Configures the server sockets used for the WAN replication's inbound WAN connections. Multiple WAN server sockets can be configured to allow configuring separate server sockets for each inbound WAN connection.",
+          "additionalProperties": false,
+          "$ref": "#/definitions/SocketEndpointConfig"
+        },
+        "memcache-server-socket-endpoint-config": {
+          "type": "object",
+          "additionalProperties": false,
+          "$ref": "#/definitions/SocketEndpointConfig"
+        },
+        "rest-server-socket-endpoint-config": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": " Controls access to Hazelcast HTTP REST API. The methods available through REST API are grouped to several REST endpoint groups, which can be specified in this section.",
+          "$ref": "#/definitions/SocketEndpointConfig",
+          "properties": {
+            "endpoint-groups": {
+              "type": "object",
+              "additionalProperties": false,
+              "description": "Enables or disables named REST endpoint groups. If a group is not listed within the rest-api configuration, then it's 'enabledByDefault' flag is used to control the behavior of the group.",
+              "properties": {
+                "CLUSTER_READ": {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "description": "Group of operations for retrieving cluster state and its version.",
+                  "properties": {
+                    "enabled": {
+                      "type": "boolean",
+                      "default": true
+                    }
+                  }
+                },
+                "CLUSTER_WRITE": {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "description": "Operations which changes cluster or node state or their configurations.",
+                  "properties": {
+                    "enabled": {
+                      "type": "boolean",
+                      "default": false
+                    }
+                  }
+                },
+                "HEALTH_CHECK": {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "description": "Group of endpoints for HTTP health checking.",
+                  "properties": {
+                    "enabled": {
+                      "type": "boolean",
+                      "default": false
+                    }
+                  }
+                },
+                "HOT_RESTART": {
```

Comment:
I think for these 4 properties we can use a "$ref" (pointing to a schema including validation props) and add separate descriptions.

Reply:
done. left out `CLUSTER_READ` as its default value is different than the others.